### DISCUSSION TITLE
Discover question about RTPS config

### DISCUSSION THREAD

**Discussion by yuan-ge** [OpenDDS/OpenDDS#4684] [Q&A] [answered]
The demo of start gives this:
[common]
DCPSGlobalTransportConfig=$file
DCPSDefaultDiscovery=DEFAULT_RTPS

[transport/the_rtps_transport]
transport_type=rtps_udp
And when I use OpenDDS in services, questionss occur when I do thee following:

1. Start all services and they connected correctly.
2. Shutdown a service A, and restart it.
3. The service A sometimes cannot connect with all services it should connect, or it just because the connect time is so long that I cannot wait for.

I have read about the RT config `ResendPeriod`, if I set this to a number in 3-5, will such problems be improved?
Another question is that since the `ResendPeriod` is 30 by default, then why is the speed relatively fast when initially establishing contact?
Thanks for your attention.

**Comment by jrw972** [comment, marked as answer]
The ResendPeriod governs the time between sending SPDP announcements which triggers discovery.
So, setting it to a value such as 3-5 seconds results in faster discovery at the cost of more periodic sends.
On startup, the SPDP is announced.
However, it has to wait for the other services to announce for discovery to complete.

See https://github.com/OpenDDS/OpenDDS/issues/2787.

**Reply by yuan-ge** [reply]
I also see the `QuickResendRatio` config, and it says "Tuning parameter that configures local SPDP resends as a fraction of the resend period." 

I think it is closely related to `ResendPeriod`, it seems that it should be multiplied by `ResendPeriod`, but I can't understand it's meaning. Would you explain it?

**Comment by jrw972** [comment]
Please look at the code to confirm.  When a new participant is discovered, the `ResendPeriod` is shorted by multiplying with the `QuickResendRatio`.  The `ResendPeriod` then gradually increases back to its original value.  Thus, if `ResendPeriod` was 30 and `QuickResendRatio` is .1, then the resend period would go down to 3 seconds when a new participant is discovered.
